Update component “checkboxes” (#116)
Zywave/booster: src/_areas/design-system/components/checkboxes.md

* Use when the number of options is easy for the user to process, generally fewer than 10.
* Checkboxes are small by its nature, and they can be hard to click or tap. To help users make easier selections, make sure the clickable area includes both the checkbox and its label.

<br>

### Gallery Checkbox

Gallery checkboxes are more prominent than standard checkboxes. They are generally used when they control the only selection the user can make on that page.

* Since Gallery Checkboxes can be significantly larger in size than Standard Checkboxes, they should generally be reserved for when that selection is the only one the user can make on that page.
* Spot illustrations are optional to use inside a Gallery Checkbox. They are used to both emphasize the selection and reinforce the contrast between the options.

<hr>

## Anatomy

For detailed documentation on our checkbox anatomy, view the [ZUI checkbox design specs](https://xd.adobe.com/view/1e451acf-dca2-47ce-8bee-99ca93473313-4b16/grid).

![Image of Standard Checkbox](/images/anatomy-standard.png)

### Standard Checkbox

The entire Checkbox can be interacted with to check or uncheck it.

> 1. Checkbox: A visual indicator for the option when it been checked or unchecked
> 2. Label: a text description of the option
> 3. Option: the combination of a checkbox and a label

Reviewer: ```suggestion
1. **Checkbox**: a visual indicator for the option when it has been checked or unchecked
2. **Label**: a text description of the option
3. **Option**: the combination of a checkbox and a label
```

![Image of Gallery Checkbox](/images/anatomy-gallery.png)

### Gallery Checkbox

The entire gallery Checkbox can be interacted with to check or uncheck it.

> 1. Checkbox: A visual indicator for the option when it been checked or unchecked
> 2. Label: a text description of the option
> 3. Spot illustration (optional): emphasizes the option and reinforces the contrast between different options

Reviewer: ```suggestion
1. **Checkbox**: a visual indicator for the option when it been checked or unchecked
2. **Label**: a text description of the option
3. **Spot illustration (optional)**: emphasizes the option and reinforces the contrast between different options
```

<hr>

## Best practices

#### Be consistent when writing Checkbox labels

* Make sure all Checkbox labels follow the same text format, such as sentence case, title case, etc.
* Avoid ending Checkbox labels with a period (.)
* Make sure all Checkbox labels follow the same pattern; which can be a sentence, a phrase, or a single word

<docs-grid columns="2">

<div>

![Keep label text format consistent](/images/group-264508.png)

<docs-do>
Keep label text format and pattern consistent
</docs-do>

</div>

<div>

![Avoid ending Checkbox labels with a period (.)](/images/2.png)

<docs-do-not>
Avoide having multiple formats such as different phrases, end in periods, and etc.
</docs-do-not>

</div>

</docs-grid>

<docs-spacer>

</docs-spacer>

#### Use positive wording for Checkbox labels

> Checkboxes universally infer affirmation, not negation. When writting the Checkbox label, make sure to use positive and active wording so that it is clear to users what *will* happen if they click on the Checkbox. 

Reviewer: ```suggestion
Checkboxes universally infer affirmation, not negation. When writing the Checkbox label, make sure to use positive and active wording so that it is clear to users what *will* happen if they click on the Checkbox. 
```

<docs-do-not>
Avoid negations in Checkbox labels, which means that the user would have to check the box in order for something not to happen.
</docs-do-not>

</div>

</docs-grid>

<docs-spacer>

</docs-spacer>

#### Lay out Checkbox lists vertically

It is recommended that check lists be displayed vertically with one checkbox per line to better distinguish each option so they are easier to scan and read. If you must use a horizontal layout, make sure to space items out abundantly so that it is easy for users to associate Checkboxes with the correct label. 

<docs-grid columns="2">

<div>

![If you have to display Checkboxes horizontally, make sure to space each checkbox out appropriately so the checkbox associates with the related label](/images/3-1.png)

<docs-do>
If you have to display Checkboxes horizontally, make sure to space each checkbox out appropriately so the checkbox associates with the related label
</docs-do>

</div>

<div>

![Avoid placing Checkboxes too close to each other as it would create confusion on associate the right checkbox to the right label](/images/3-2.png)

<docs-do-not>

> Avoid placing Checkboxes too close to each other as it would create confusion on associate the right checkbox to the right label

Reviewer: ```suggestion
Avoid placing Checkboxes too close to each other as it would create confusion and not associate the right checkbox to the right label
```

</docs-do-not>

</div>

</docs-grid>

<docs-spacer>

</docs-spacer>

> #### Use Tri-State Checkbox when a list is partially selected

Reviewer: ```suggestion
#### Use a Tri-State Checkbox when a list is partially selected
```

> An indeterminate state is required when sub-options are grouped under a parent Checkbox. The indeterminate state represents a parent Checkbox that is partially checked due to some of its sub-options only being checked.

Reviewer: ```suggestion
The indeterminate state is required to represent a partially checked parent Checkbox when some of its sub-options are checked.
```
